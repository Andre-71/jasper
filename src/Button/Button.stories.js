import React from "react"
import { storiesOf } from "@storybook/react"
import Button from "."

storiesOf("Button").add("Default", () => <Button> Hello </Button>)
