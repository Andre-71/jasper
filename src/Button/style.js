import styled from "styled-components"

export const StyledButton = styled.button`
  background: ${props => props.theme.colors.primary ? props.theme.colors.primary : "linear-gradient(111.34deg, #D418B6 0%, #F1185E 100%)"};
  border-radius: 100px;
  border: none;
  padding: 12px 22px;
  cursor: pointer;
  color: white;
  font-size: 16px;

  &.secondary {
    background: transparent;
    border: 2px solid #f4f4f4;
    padding: 10px 20px;
    color: white;
  }

  &.nude {
    background: transparent;
  }

  &.bold {
    font-weight: bold;
  }

  &:hover {
    opacity: 0.75;
  }
`
