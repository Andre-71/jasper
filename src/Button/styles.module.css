.all {
  background: #cb0058;
  box-shadow: 0px 2px 5px rgba(0, 0, 0, 0.2);
  border-radius: 100px;
  border-color: transparent;
  border-width: 1px;
  padding: 10px 20px;
  cursor: pointer;
  color: white;
  font-size: 16px;
}

.all:hover {
  opacity: 0.75;
}
