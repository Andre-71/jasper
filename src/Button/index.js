import React from "react"

import PropTypes from "prop-types"
import { StyledButton } from "./style"

const Button = props => {
  if (props.href) {
    if (props.download) {
      return (
        <a href={props.href} download>
          <StyledButton {...props}>{props.children}</StyledButton>
        </a>
      )
    } else {
      return (
        <a href={props.href}>
          <StyledButton {...props}>{props.children}</StyledButton>
        </a>
      )
    }
  }
  return <StyledButton {...props}>{props.children}</StyledButton>
}

Button.propTypes = {
  href: PropTypes.string,
  children: PropTypes.string,
}
Button.defaultProps = {
  type: "button",
}

export default Button
