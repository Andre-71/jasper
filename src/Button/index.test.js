import React from 'react'
import Button from '.'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  let tree = renderer
    .create(<Button>Hello</Button>)
    .toJSON()
  expect(tree).toMatchSnapshot()

  tree = renderer
    .create(<Button href="https://www.compfest.id/">Hi</Button>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
