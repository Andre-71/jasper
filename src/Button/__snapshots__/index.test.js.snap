// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="all"
  type="button"
>
  Hello
</button>
`;

exports[`renders correctly 2`] = `
<a
  href="https://www.compfest.id/"
>
  <button
    className="all"
    href="https://www.compfest.id/"
    type="button"
  >
    Hi
  </button>
</a>
`;
